src/codegen.cpp

```diff
@@ -1563,7 +1547,9 @@ void *jl_get_llvmf_decl(jl_method_instance_t *linfo, size_t world, bool getwrapp
     }
 
     // compile this normally
-    jl_llvm_functions_t decls = jl_compile_for_dispatch(&linfo, world);
+    if (linfo->inferred == NULL)
+        linfo->inferred = jl_nothing;
```

Comment:
It's generally invalid to change the `->inferred` field, only `jl_set_method_inferred` should be doing that.

Reply:
Would it be better to handle this in `jl_compile_linfo`? It's not clear to me why `jl_compile_linfo` does `goto locked_out` if `li->inferred` is NULL.